membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -69,11 +73,13 @@ class AttributesFromZuora(implicit val executionContext: ExecutionContext, syste
           if(userHasDigiSub(subscriptions)) Future.successful(\/.right[String, List[GiftSubscriptionsFromIdentityIdRecord]](Nil))
           else giftSubscriptionsForIdentityId(identityId)
         )
-        giftAttributes = giftSubscriptions.headOption.map(record => ZuoraAttributes(identityId, DigitalSubscriptionExpiryDate = Some(record.TermEndDate)))
-        maybeZAttributes <- EitherT(AttributesMaker.zuoraAttributes(identityId, subscriptions, paymentMethodForPaymentMethodId, forDate).map(\/.right[String, Option[ZuoraAttributes]]))
+        maybeGiftAttributes = giftSubscriptions
+          .sortWith((first, second) => first.TermEndDate.isAfter(second.TermEndDate)).headOption
```

Comment:
sorting always confuses me, also not needed here
https://www.scala-lang.org/api/2.12.12/scala/collection/TraversableOnce.html#maxBy[B](f:A=%3EB):A

so something like this, although haven't checked the joda API to get the proper method name

```suggestion
        maybeGiftAttributes = if (giftSubscriptions.isEmpty) 
          None
        else
          Some(giftSubscriptions.maxBy(_.TermEndDate.toMillisLong))
```